supervisor/agent/agent.go

```diff
@@ -0,0 +1,64 @@
+package agent
+
+import "github.com/open-telemetry/opamp-go/protobufs"
+
+type Settings struct {
+	// Capabilities are the list of capabilities that this agent supports.
+	// See `protobufs.AgentCapabilities'.
+	Capabilities []string `koanf:"capabilities"`
```

Comment:
Can we use "mapstructure" field tags instead of "koanf"? That's what we do in Collector codebase and would be nice to be consistent here.